parlai/agents/transformer/decoder_only.py

```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/env python3
+
+# Copyright (c) Facebook, Inc. and its affiliates.
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+from typing import Optional
+
+from parlai.agents.transformer.transformer import add_common_cmdline_args
+from parlai.core.opt import Opt
+from parlai.core.params import ParlaiParser
+from parlai.core.torch_generator_agent import TorchGeneratorAgent
+from parlai.utils.logging import logging
+from parlai.utils.misc import recursive_getattr
+from parlai.utils.torch import padded_tensor
+
+from .modules import (
+    PassThroughEncoder,
+    TransformerDecoderOnly,
+    TransformerGeneratorModel,
+)
+
+
+class DecoderOnlyAgent(TorchGeneratorAgent):
```

Comment:
nit, maybe just `transformer/decoder`?

Reply:
It'll be a liiiittle confusing when navigating through files, since there's also a `transformer/modules/decoder.py`. But I agree with you that we should prioritize the user-facing model name! Will change.